File: Chavah/Models/Artist.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace BitShuva.Models
{
    public class Artist
    {
        public Artist()
        {
            this.Images = new List<string>();
        }

        public string Id { get; set; }
        public string Name { get; set; }
        public List<string> Images { get; set; }
        public string Bio { get; set; }
    }
}

File: Chavah/wwwroot/js/Models/Artist.ts
namespace BitShuva.Chavah {
    export class Artist implements Server.Artist {
        name: string;
        images: string[];
        bio: string;

        isSaving = false;

        constructor(serverObj?: Server.Artist) {
            if (!serverObj) {
                serverObj = Artist.createDefaultServerObj();
            }
            angular.merge(this, serverObj);
        }

        updateFrom(serverObj: Server.Artist) {
            angular.merge(this, serverObj);
        }

        // tslint:disable-next-line:member-ordering
        static createDefaultServerObj(): Server.Artist {
            return {
                bio: "",
                images: [],
                name: "",
            };
        }
    }
}
